osu.Game/Screens/Select/BeatmapPlaylist.cs

```diff
@@ -0,0 +1,348 @@
+
+                return base.OnMouseMove(e);
+            }
+
+            protected override bool OnMouseDown(MouseDownEvent e)
+            {
+                if (!e.IsPressed(MouseButton.Left))
+                    return false;
+
+                if (dragHandle.IsHovered)
+                    removeButton.Hide();
+
+                return base.OnMouseDown(e);
+            }
+
+            protected override bool OnMouseUp(MouseUpEvent e)
+            {
+                if (!suppressHover)
+                {
+                    showHoverElements(false);
+                    setHighlighted(false);
+                }
+                else
+                {
+                    removeButton.Show();
+                }
+
+                return base.OnMouseUp(e);
+            }
+
+            private void showHoverElements(bool show)
+            {
+                if (show)
+                {
+                    removeButton.Show();
+                    dragHandle.Show();
+                }
+                else
+                {
+                    removeButton.Hide();
+                    dragHandle.Hide();
+                }
+            }
+
+            private void setHighlighted(bool selected)
+            {
+                var colour = selected ? selectedColour : backgroundColour;
+                background.FadeColour(colour);
+                gradient.FadeColour(ColourInfo.GradientHorizontal(colour, colour.Opacity(0.5f)));
+            }
+
+            private class DragHandle : OsuClickableContainer
+            {
+                public DragHandle()
+                {
+                    RelativeSizeAxes = Axes.Y;
+                    Width = 25;
+                    Alpha = 0;
+                    Child = new SpriteIcon
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                        Size = new Vector2(12),
+                        Icon = FontAwesome.Solid.Bars,
+                        Margin = new MarginPadding { Left = 5, Top = 2 }
+                    };
+                }
+
+                public override bool HandlePositionalInput => IsPresent;
+
+                public override void Show() => this.FadeIn(fade_duration);
+
+                public override void Hide() => this.FadeOut(fade_duration);
+            }
+
+            private class RemoveButton : OsuClickableContainer
+            {
+                private readonly SpriteIcon icon;
+                public bool Depressed;
+
+                public RemoveButton()
+                {
+                    Alpha = 0;
+                    Child = icon = new SpriteIcon
+                    {
+                        Colour = Color4.White,
+                        Icon = FontAwesome.Solid.MinusSquare,
+                        Size = new Vector2(14),
+                    };
+                }
+
+                public override void Show() => this.FadeIn(fade_duration);
```

Comment:
We might want to move these fading + duration things to the base class and set duration value here. However, this is probably out of scope for this pull request.